ssr/src/page/wallet/mod.rs

```diff
@@ -116,43 +131,122 @@ fn TokensFetch() -> impl IntoView {
         </Suspense>
     }
 }
-
+#[derive(Params, PartialEq)]
+struct WalletParams {
+    id: String,
+}
 #[component]
 pub fn Wallet() -> impl IntoView {
+    let params = use_params::<WalletParams>();
+    let param_principal = move || {
+        params.with(|p| {
+            let WalletParams { id, .. } = p.as_ref().ok()?;
+            Principal::from_text(id).ok()
+        })
+    };
+
+    view! {
+        {
+            move || {
+                match param_principal(){
+                    Some(principal) => view! {<WalletImpl principal/>},
+                    None => view! {
+                        <AuthCansProvider let: cans>
+                        {move ||{
+                                view!{<Redirect path=format!("/wallet/{}", cans.user_principal())/>}
+                            }}
+                        </AuthCansProvider>
+                    }
+                }
+            }
+        }
+    }
+}
+#[component]
+pub fn WalletImpl(principal: Principal) -> impl IntoView {
     let (is_connected, _) = account_connected_reader();
+    let param_principal = move || principal;
 
     let auth_cans = authenticated_canisters();
-    let balance_fetch = auth_cans.derive(
-        || (),
-        |cans_wire, _| async move {
-            let cans = cans_wire?.canisters()?;
-            let user = cans.authenticated_user().await;
+    let balance_fetch = auth_cans.derive(param_principal, |cans_wire, principal| async move {
```

Comment:
Do you need an authenticated canister for this?
If not, use unauth_canisters with create_resource

Reply:
do i do the same for history fetch too?